src/main/kotlin/day15/data_source/MockTodoDatasourceImpl.kt

```
@@ -1,17 +1,27 @@
-    override suspend fun getCompletedTodos(): List<Todo> {
-        return json.filter { it.completed }
+    override suspend fun getCompletedTodos(): List<Todo> = withContext(Dispatchers.IO) {
+        json.filter { it.completed }
     }
 
-    private val json = Json.decodeFromString<List<Todo>>(File("JsonList.txt").readText())
+    private val json: List<Todo> by lazy {
+        try {
+            val file = File("JsonList.txt")
+            Json.decodeFromString(file.readText())
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)